lib/apiservers/portlayer/restapi/handlers/interaction_handlers.go

```diff
@@ -118,25 +134,59 @@ func (i *InteractionHandlersImpl) ContainerSetStdinHandler(params interaction.Co
 }
 
 func (i *InteractionHandlersImpl) ContainerGetStdoutHandler(params interaction.ContainerGetStdoutParams) middleware.Responder {
-	log.Printf("Attempting to get ssh session for container %s stdout", params.ID)
-	sshConn, err := i.attachServer.Get(context.Background(), params.ID, interactionTimeout)
-	if err != nil {
+	var ctxDeadline time.Time
+	var timeout time.Duration
+
+	// Calculate the timeout for the attach if the caller specified a deadline
+	if params.Deadline != nil {
+		ctxDeadline = time.Time(*params.Deadline)
+		timeout = ctxDeadline.Sub(time.Now())
+		log.Printf("Attempting to get ssh session for container %s stdout with deadline %s", params.ID, ctxDeadline.Format(time.UnixDate))
+		if timeout < 0 {
+			e := &models.Error{Message: fmt.Sprintf("Deadline for stdin already passed for container %s", params.ID)}
+			return interaction.NewContainerGetStdoutInternalServerError().WithPayload(e)
+		}
+	} else {
+		log.Printf("Attempting to get ssh session for container %s stdout", params.ID)
+		timeout = interactionTimeout
+	}
+
+	sshConn, err := i.attachServer.Get(context.Background(), params.ID, timeout)
+	if err != nil || sshConn == nil {
 		e := &models.Error{Message: fmt.Sprintf("No stdout found for %s", params.ID)}
 		return interaction.NewContainerGetStdoutNotFound().WithPayload(e)
 	}
 
+	log.Printf("* Returning ContainerOutputHandler...")
 	detachableOut := NewFlushingReader(sshConn.Stdout())
 	return NewContainerOutputHandler("stdout").WithPayload(detachableOut, params.ID)
 }
 
 func (i *InteractionHandlersImpl) ContainerGetStderrHandler(params interaction.ContainerGetStderrParams) middleware.Responder {
-	log.Printf("Attempting to get ssh session for container %s stderr", params.ID)
-	sshConn, err := i.attachServer.Get(context.Background(), params.ID, interactionTimeout)
+	var ctxDeadline time.Time
+	var timeout time.Duration
+
+	// Calculate the timeout for the attach if the caller specified a deadline
+	if params.Deadline != nil {
+		ctxDeadline = time.Time(*params.Deadline)
+		timeout = ctxDeadline.Sub(time.Now())
```

Comment:
And you're doing the inverse here.  This just requires a paramter to set the timeout.  Deadlines are nice if you need somethign done by a certain wall clock time but i don't see that requirement here.  Just pass a timeout.

Also, the timeout duration itself looks like it's coming from a constant.  So I'm kind of wondering why we need this change at all.  Am I missing something?